src/main/java/com/eppo/sdk/helpers/bandit/BanditEvaluator.java

```diff
@@ -0,0 +1,86 @@
+package com.eppo.sdk.helpers.bandit;
+
+import com.eppo.sdk.dto.EppoAttributes;
+import com.eppo.sdk.dto.EppoValue;
+import com.eppo.sdk.dto.ShardRange;
+import com.eppo.sdk.dto.Variation;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class BanditEvaluator {
+
+    public static List<Variation> evaluateBanditVariations(
+      String banditKey,
+      String modelName,
+      Map<String, EppoAttributes> actions,
+      String subjectKey,
+      EppoAttributes subjectAttributes,
+      int subjectShards
+    ) {
+        BanditModel model = BanditModelFactory.build(modelName);
+        List<String> shuffledActions = shuffleVariations(actions.keySet(), banditKey, subjectKey);
+        Map<String, Float> actionWeights = model.weighActions(actions, subjectAttributes);
+        return generateVariations(shuffledActions, actionWeights, subjectShards);
+    }
+
+    private static List<String> shuffleVariations(Set<String> actionKeys, String banditKey, String subjectKey) {
+        return actionKeys
+          .stream()
+          .sorted(Comparator.comparingInt(actionKey -> hashToPositiveInt(subjectKey, banditKey, actionKey)))
+          .collect(Collectors.toList());
+    }
+
+    private static int hashToPositiveInt(String actionKey, String banditKey, String subjectKey) {
+        try {
+            // Create MD5 Hash
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            md.update(actionKey.getBytes(StandardCharsets.UTF_8));
+            md.update(banditKey.getBytes(StandardCharsets.UTF_8));
+            md.update(subjectKey.getBytes(StandardCharsets.UTF_8));
+            byte[] digest = md.digest();
+
+            // Use the first 4 bytes to create an int and ensure it's positive
```

Comment:
We make sure its positive so we don't overflow when subtracting when the comparator is used to "sort" (shuffle) the actions.

Reply:
What is the consequence of collisions?

Reply:
Good question! We're using this to `sort()` a `stream()`, based on the [docs](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#sorted--), it's only stabled for ordered streams. Since we're streaming a Set, the order is not guaranteed. So, a collision could result in the actions flipping order each time. Although unlikely, to guard against this, we can tie break with the action name. This strategy will also make it easy to ensure consistent sorting in other languages as well.